src/dao.ts

```diff
@@ -171,58 +160,39 @@ export class DAO implements IStateful<IDAOState> {
       ${DAO.fragments.DAOFields}
      `
 
-    const itemMap = (item: any): IDAOState => {
-      if (item === null) {
-        throw Error(`Could not find a DAO with id ${this.id}`)
-      }
-      const reputation = new Reputation(this.context, item.nativeReputation.id)
-      const token = new Token(this.context, item.nativeToken.id)
-      const state = {
-        address: item.id,
-        dao: this,
-        id: item.id,
-        memberCount: Number(item.reputationHoldersCount),
-        name: item.name,
-        numberOfBoostedProposals: Number(item.numberOfBoostedProposals),
-        numberOfPreBoostedProposals: Number(item.numberOfPreBoostedProposals),
-        numberOfQueuedProposals: Number(item.numberOfQueuedProposals),
-        register: item.register,
-        reputation,
-        reputationTotalSupply: new BN(item.nativeReputation.totalSupply),
-        token,
-        tokenName: item.nativeToken.name,
-        tokenSymbol: item.nativeToken.symbol,
-        tokenTotalSupply: item.nativeToken.totalSupply
-      }
-      this.setState(state)
-      return state
-    }
+    const itemMap = (item: any) => DAO.itemMap(this.context, item)
+
     return this.context.getObservableObject(query, itemMap, apolloQueryOptions)
   }
 
-  /*
-   * return the nativeReputation of the DAO
-   * @returns an (Observable) that returns a Reputation instance
-   */
   public nativeReputation(): Observable<Reputation> {
-    return this.state().pipe(first()).pipe(map((r) => r.reputation))
+    return this.state().pipe(first()).pipe(map((r) => r.reputation.entity))
+  }
+
+  public ethBalance(): Observable<BN> {
+    return this.context.ethBalance(this.id)
   }
 
-  public schemes(
-    options: ISchemeQueryOptions = {},
+  public plugins(
+    options: IPluginQueryOptions = {},
     apolloQueryOptions: IApolloQueryOptions = {}
-  ): Observable<SchemeBase[]> {
+  ): Observable<ProposalPlugin[]> {
     if (!options.where) { options.where = {}}
     options.where.dao = this.id
-    return Scheme.search(this.context, options, apolloQueryOptions)
+    return ProposalPlugin.search(this.context, options, apolloQueryOptions)
   }
 
-  public async scheme(options: ISchemeQueryOptions): Promise<SchemeBase> {
-    const schemes = await this.schemes(options).pipe(first()).toPromise()
-    if (schemes.length === 1) {
-      return schemes[0]
+  /* TODO
+  public proposalplugins() {
```

Comment:
I actually had this implemented already but did not have the query to get only the schemes that can create proposals

Reply:
How would be the query to retrieve only proposalPlugins look like?

Reply:
We'd want to use the proposal type names, right?

Reply:
Actually, sorry, this is more so what we want:
```ts
const ProposalPlugins = {
  GenericScheme,
  SchemeRegistrar,
  ContributionReward,
  ContributionRewardExt,
  Competition
}

type ProposalPluginName = keyof typeof ProposalPlugins

const Plugins = {
  ...ProposalPlugins,
  ReputationFromToken
}

type PluginName = keyof typeof Plugins
```

In this ^^^ it's assuming ReputationFromToken is fixed to be a class that derives from Plugin

Reply:
Right, okay. I'll do this refactor but I'll leave the ReputationFromToken out of the Plugins object for now, since its interface does not comply with the Plugin interface. This is because the Plugin interface has a coreState field, a state method (ReputationFromToken); and the original codebase did not have a ItemMap or Search implementation